### PR TITLE
feat: PascalCase resource name generation check

### PR DESCRIPTION
**Pull request by Fecony**
## Description
This PR is fixing test issue where tests fail because new feature is not formatting user input properly.
Change flow to check if name is valid, and then throw error if it is not.

## Changes
- Remove `resourceNameToClassName` method
- Add 3 new methods, 1 exported `checkResourceNameIsValid`
- Throw error if resource name is not valid PascalCase to force user to use correct naming
- Fix generator tests to use valid PascalCase name for files
- Add new tests

## Checks
- [x] Project Builds
- [x] Project passes tests and checks

### CODE REVIEW

File: packages/cli/src/common/filenames.ts [hunk 1]
```diff
@@ -1,5 +1,38 @@
 import * as inflected from 'inflected'
+import { logger } from '../services/logger'
+import Brand from './brand'
 
 export function classNameToFileName(name: string): string {
   return inflected.dasherize(inflected.underscore(name))
 }
+
+export async function checkResourceNameIsValid(name: string): Promise<void> {
+  logger.info(Brand.mellancholize('Checking if resource has valid name...'))
```
Comment (davidverdu) [inline packages/cli/src/common/filenames.ts:10]: I would't display this message in the console. It only makes sense displaying on errors

**Review by davidverdu — CHANGES_REQUESTED** [review]
I had to rebase main because there was an error launching `lerna run integration/cli` . Now the PR has some conflicts. To accept this PR you need to solve the conflicts first. I also left a minor comment. 

Before submitting the PR, please verify that `lerna run test` and `lerna run integration/cli --stream` run without errors (these scripts doesn't need having a AWS account)

File: packages/cli/src/common/filenames.ts [hunk 2]
```diff
@@ -3,3 +3,32 @@ import * as inflected from 'inflected'
 export function classNameToFileName(name: string): string {
   return inflected.dasherize(inflected.underscore(name))
 }
+
+export async function checkResourceNameIsValid(name: string): Promise<void> {
```
Comment (rdiaz82) [inline packages/cli/src/common/filenames.ts:7]: one question, Why is this function async? I don't see any async call inside the method

**Review by rdiaz82 — APPROVED** [review]
In general LGTM just a question to clarify!!

**Review by charlietfe — APPROVED** [review]
In general LGTM, nothing to add. Thanks for contributing @Fecony

File: packages/cli/src/common/filenames.ts (hunk 2 above)
Comment (Fecony) [reply packages/cli/src/common/filenames.ts]: Hey, @rdiaz82 yeap, looks like it is not required, I only used it to do fancy `.then/.catch` in test file 😸

**Review by davidverdu — APPROVED** [review]
LGTM

**Pull request merged**